===================================================================
RCS file: /home/sdlweb/libsdl.org/cvs/SDL12/src/main/macosx/SDLMain.m,v
retrieving revision 1.15
retrieving revision 1.16
diff -u -r1.15 -r1.16
--- SDL12/src/main/macosx/SDLMain.m	2005/08/22 14:18:15	1.15
+++ SDL12/src/main/macosx/SDLMain.m	2006/01/01 23:45:52	1.16
@@ -190,7 +190,7 @@
 }
 
 /* Replacement for NSApplicationMain */
-static void CustomApplicationMain (argc, argv)
+static void CustomApplicationMain (int argc, char **argv)
 {
     NSAutoreleasePool	*pool = [[NSAutoreleasePool alloc] init];
     SDLMain				*sdlMain;
@@ -251,8 +251,8 @@
     if (gCalledAppMainline)  /* app has started, ignore this document. */
         return FALSE;
 
-    unsigned buflen = [filename lengthOfBytesUsingEncoding:NSUTF8StringEncoding] + 1;
-    char *arg = (char *) malloc(buflen);
+    const char *temparg = [filename UTF8String];
+    char *arg = (char *) malloc(strlen(temparg) + 1);
     if (arg == NULL)
         return FALSE;
 
@@ -264,15 +264,10 @@
     }
     gArgv = newargv;
 
-    BOOL rc = [filename getCString:arg maxLength:buflen encoding:NSUTF8StringEncoding];
-    if (!rc)
-        free(arg);
-    else
-    {
-        gArgv[gArgc++] = arg;
-        gArgv[gArgc] = NULL;
-    }
-    return rc;
+    strcpy(arg, temparg);
+    gArgv[gArgc++] = arg;
+    gArgv[gArgc] = NULL;
+    return TRUE;
 }
 
 
